packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1837,11 +1839,28 @@ export interface ApplyCloudFormationInitOptions {
   readonly includeUrl?: boolean;
 
   /**
-  * Include --role argument when running cfn-init and cfn-signal commands
-  *
-  * This will be the IAM instance profile attached to the EC2 instance
-  *
-  * @default false
-  */
+   * Include --role argument when running cfn-init and cfn-signal commands
+   *
+   * This will be the IAM instance profile attached to the EC2 instance
+   *
+   * @default false
+   */
   readonly includeRole?: boolean;
+  /**
+   * Include --http-proxy argument when running cfn-init and cfn-signal commands
+   *
+   * An HTTP proxy (non-SSL). Use the following format: http://user:password@host:port
+   *
+   * @default
+   */
+  readonly httpProxy?: string;
+
+  /**
+   * Include --https-proxy argument when running cfn-init and cfn-signal commands
+   *
+   * An HTTPS proxy. Use the following format: https://user:password@host:port
+   *
+   * @default
+   */
+  readonly httpsProxy?: string;
```

Comment:
as these are secrets, I would like to propose an alternative way:
An interface like this: 
```
interface IProxyConfig {
protocol: string,
user: string,
password: ISecret,
host: string,
port: number
}

Reply:
Finally getting back to this. Currently figuring out how to deal with circular dependencies between packages.

`aws-secretsmanager` has a dependency on the `aws-ec2` package, causing issues when doing a `yarn build` within the `aws-ec2` package after adding secrets manager.

Trying to figure out how to handle this if possible

Reply:
Articulated this in #19479